### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by misantron**
- Unit tests for Logger, Request and Session classes
- Logger refactoring (use SplFileObject instead of file_get_contents/file_put_contents)
- Bug fixes

### CODE REVIEW

**Comment by misantron** [comment]
Waiting for code review

File: src/WebTerminal/Logger.php [hunk 1]
```diff
@@ -4,18 +4,28 @@
...
+    private $file;
 
     /**
      * Logger constructor.
-     * @param $filePath
+     *
+     * @param string $filePath
      */
-    public function __construct($filePath)
+    public function __construct(string $filePath)
```
Comment (captainspain) [inline src/WebTerminal/Logger.php:22]: This constructor should not throw an Exception if file not found it should create one.

File: src/WebTerminal/Logger.php [hunk 2]
```diff
@@ -57,13 +65,16 @@ public function setData(array $data): self
      */
     public function save(): void
     {
+        $file = $this->file;
         $data = $this->data;
-        file_put_contents($this->filePath, json_encode($data, JSON_FORCE_OBJECT));
+
+        $file->rewind();
```
Comment (captainspain) [inline src/WebTerminal/Logger.php]: The rewind function is triggering a Warning: `SplFileObject::rewind(): stream does not support seeking`

**Review by captainspain — CHANGES_REQUESTED** [review]
Thanks a lot for this PR.
Looks good to me, only 2 minor issues. If you can fix them I will merge it :)

File: src/WebTerminal/Logger.php (hunk 1 above)
Comment (misantron) [reply src/WebTerminal/Logger.php:22]: Ok but what should we do if the file is not readable?

Comment (captainspain) [reply src/WebTerminal/Logger.php:22]: @misantron create a new file. If that's not possible then throw one

Comment (misantron) [reply src/WebTerminal/Logger.php:22]: @captainspain rollback SplFileObject usage. We get more problems than we solve.

**Review by captainspain — APPROVED** [review]

**Comment by captainspain** [comment]
Thanks a lot @misantron

**Pull request merged**